FOV is Dead (Long Live FOV)

LemonInTheDark opened the pull request:
## About The Pull Request

FOV as it is currently implemented is incompatible* with wallening.
I'm doin wallening, so we gotta redo things here.

The issue is the masking of mobs. Wallening relies on sidemap (layering based off physical position), which only works on things on the same plane (because planes are basically sheets we render down onto)
So rather then masking mobs, let's reuse the masking idea from old fov, and use it to cut out a bit of the game render plane, and blur/over-saturate the bit that's masked out.

My hope is this makes things visible in light, but not as much in darkness, alongside making more vivid shit more easily seen (just like real life)

Here's some videos, what follows after is the commits I care about (since I had to rip a bunch of planes to nothing, so the files changed tab might be a bit of a mess)

Oh also I had to remove the darkness pref since the darkness is doing a lot of the heavy lifting now. I'm sorry.

Edit:
NEW FOV SPRITES! Thanks dongle your aviator glasses will guide us to a better future.

https://github.com/tgstation/tgstation/assets/58055496/afa9eeb8-8b7b-4364-b0c0-7ac8070b5609

https://github.com/tgstation/tgstation/assets/58055496/0eff040c-8bf1-47e4-a4f3-dac56fb2ccc8

## Commits I Care About

[Implements something like fov, but without the planes as layers hell](https://github.com/tgstation/tgstation/commit/a604c7b1c8d74cd27af4d806d85892c1f7e35ba8)

Rather then masking out mobs standing behind us, we use a combo color matrix and blur filter to make the stuff covered by fov harder to see.

We achive this by splitting the game plane into two, masking both by fov (one normally and one inversely), and then applying effects to one of the two.

I want to make the fov fullscreens more gradient, but as an effect this is a good start

[Removes WALL_PLANE_UPPER by adding a WALL_PLANE overlay to material walls (init cost comes here)](https://github.com/tgstation/tgstation/commit/25489337392f708cb337fbf05a2329eacdfc5346)

@Mothblocks see this. comment in commit explains further but uh, we need to draw material walls to the light mask plane so things actually can be seen on them, but we can't do that and also have them be big, so they get an overlay. Sorry, slight init time bump, about 0.5 seconds. I can kill it with wallening.

[Moves SEETHROUGH_PLANE above ABOVE_GAME_PLANE](https://github.com/tgstation/tgstation/commit/beec4c00e01d34a04fba7c2bb98a9b70d27ead82)

I don't think it actually wants to draw here
@Time-Green I think this was you so pinging for opinion

[Resprites FOV masks to be clean (and more consistent)](https://github.com/tgstation/tgstation/pull/80062/commits/f02ad13696b3b17658af612c62848b48609d785d)
[f02ad13](https://github.com/tgstation/tgstation/pull/80062/commits/f02ad13696b3b17658af612c62848b48609d785d)

This is 100% donglesplonge's work, he's spent a week or so going back and forth with me sharpening these to a mirror shine, real chill

## Why It's Good For The Game

Walls are closing in

## Changelog
:cl: LemonInTheDark, Donglesplonge
image: Redoes fov "mask" sprites. They're clean, have a very pleasant dithering effect, and look real fuckin good! 
del: Changed FOV, it no longer hides mobs, instead it blurs the hidden area, and makes it a bit darker/oversaturated
/:cl:

###### * It's technically possible if we start using render targets to create 2 sets of sources but that's insane and we aren't doing it

LemonInTheDark committed:
Implements something like fov, but without the planes as layers hell

Rather then masking out mobs standing behind us, we use a combo color
matrix and blur filter to make the stuff covered by fov harder to see.

We achive this by splitting the game plane into two, masking both by fov
(one normally and one inversely), and then applying effects to one of
the two.

I want to make the fov fullscreens more gradient, but as an effect this
is a good start

LemonInTheDark committed:
Removes the first fov masked plane

LemonInTheDark committed:
Removes GAME_PLANE_UPPER

LemonInTheDark committed:
Removes WALL_PLANE_UPPER by adding a WALL_PLANE overlay to material walls (init cost comes here)

LemonInTheDark committed:
Removes GAME_PLANE_UPPER_FOV_HIDDEN

LemonInTheDark committed:
Moves SEETHROUGH_PLANE above ABOVE_GAME_PLANE

I don't think it actually wants to draw here, and this also lets me nuke
another plane master which is good for performance/cleanliness

LemonInTheDark committed:
Renumbers the planes in _layers to make debugging easier

LemonInTheDark committed:
fixes up names/descs:

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into prespective-shift

LemonInTheDark committed:
Nukes old pms, cleans descs, makes the minerals bs actually work (also yeets moth's unneeded mineral smoothing stuff

LemonInTheDark committed:
whoop

LemonInTheDark committed:
whoop

LemonInTheDark committed:
removes preference since we can no longer use it

LemonInTheDark committed:
lint bitch

LemonInTheDark commented:
it's basically like, should seethrough stuff really render above like, plasma fires or cleaning bubbles

LemonInTheDark commented:
Unrelated, donglesplonge is working on some new fov states for me (the existing ones are a little rough on the edges, literally)

Ghommie reviewed (commented):
Definitely an improvement and it removes some of the cursed issues with the current FOV, though I feel like the cone is way too dark compared to the usually brighter environment and that the blurring should be bumped up a bit.

Also yeah, the edges are rough with too little transition, good thing someone is working on it.

DATA-xPUNGED commented:
The HUGE darkness snapping around the screen feels very distracting. It's supposed to be where i can't see but it actually calls my eyes to it.

scrunklershrungle932 commented:
the darkness going around the sides of the screen when you turn seems a lil bit distracting, but I still think it's nice that the eyes everyone has in the back of their head are going away.
I think this is a good change.

LemonInTheDark commented:
> Definitely an improvement and it removes some of the cursed issues with the current FOV, though I feel like the cone is way too dark compared to the usually brighter environment and that the blurring should be bumped up a bit.

My goal was sorta like looking through a foggy glass. If I make it too much brighter the oversaturation just starts to make things look like hotdog world (also the darkness is an upside cause it makes dark areas darker, which is good imo).

I think the blurring is alright as of now, tho idk you might be able to pump it. it's pretty easy to play around with, I like how it is but if you wanna mutz with it that's fine too.

Mothblocks commented:
Make sure to document the 0.5s loss on https://github.com/tgstation/dev-cycles-initiative/issues

LemonInTheDark commented:
> :(

what'd I do

ATH1909 commented:
how much of this mechanic can be circumvented by just cranking up your screen brightness?

Mothblocks commented:
you amde init slower

LemonInTheDark committed:
whop

LemonInTheDark committed:
Resprites FOV masks to be clean (and more consistent)

This is 100% donglesplonge's work, he's spent a week or so going back
and forth with me sharpening these to a mirror shine, real chill

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into prespective-shift
